eclair-node/src/main/scala/fr/acinq/eclair/Boot.scala

```diff
@@ -84,6 +97,46 @@ object Boot extends App with Logging {
     }
   }
 
+  /**
+   * Generates the cookie file and return the password
+   */
+  def generateCookie(pathString: String, groupReadable: Boolean): String = {
+    val hexPassword = randomBytes32().toHex
+
+    val path = Path.of(pathString)
+    Files.deleteIfExists(path)
+    Files.createFile(path)
+
+    val posixView = Files.getFileAttributeView(path, classOf[PosixFileAttributeView])
+    val aclView = Files.getFileAttributeView(path, classOf[AclFileAttributeView])
+    if (posixView != null) {
+      // Change permissions for the .cookie file on Linux/Mac OS
+      if (groupReadable) {
+        posixView.setPermissions(PosixFilePermissions.fromString("rw-r-----"))
+      } else {
+        posixView.setPermissions(PosixFilePermissions.fromString("rw-------"))
+      }
+    } else if (aclView != null) {
+      // Change permissions for the .cookie file on Windows
+      val aclEntry = AclEntry.newBuilder()
+        .setPermissions(Set.from(AclEntryPermission.values()).asJava)
+        .setPrincipal(aclView.getOwner)
+        .setType(AclEntryType.ALLOW)
+        .build()
+      // setAcl() replaces all acl entries for the file.
+      // That means setting acl with an list containing only an entry that gives access to the owner, everyone else lose access to the file.
+      aclView.setAcl(List(aclEntry).asJava)
+      if (groupReadable) {
+        logger.warn("could not make the .cookie file group readable")
```

Comment:
This shouldn't be a warning, it should be an exception.
We should note in the `reference.conf` that this field cannot be set to `true` on windows.